### PR TITLE
xtensa: use CONFIG_ARCH_FPU macro for coprocossor and add up_fpucmp

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
## Summary
In xtensa,  use CONFIG_ARCH_FPU for up_fpucmp.

For consist with other archs

## Impact

No. 

## Testing
Testing ostest in xtensa arch board.

### CODE REVIEW

**Review by acassis — APPROVED** [review]

**Comment by acassis** [comment]
the FPU on ostest is failing:
xtensa-esp32-elf-ld: /github/workspace/sources/nuttx/staging/libapps.a(fpu.c.github.workspace.sources.apps.testing.ostest.o): in function `fpu_task':
/github/workspace/sources/apps/testing/ostest/fpu.c:214: undefined reference to `up_fpucmp'
xtensa-esp32-elf-ld: /github/workspace/sources/apps/testing/ostest/fpu.c:227: undefined reference to `up_fpucmp'

File: arch/xtensa/Kconfig [hunk 1]
```diff
@@ -13,6 +13,7 @@ config ARCH_CHIP_ESP32
 	bool "Espressif ESP32"
 	select ARCH_FAMILY_LX6
 	select XTENSA_HAVE_INTERRUPTS
+	select ARCH_HAVE_FPU
```
Comment (xiaoxiang781216) [inline arch/xtensa/Kconfig:16]: @Ouss4 has esp32 FPU?

File: arch/xtensa/include/irq.h [hunk 2]
```diff
@@ -123,7 +123,7 @@
 #  define _REG_CP_START  _REG_OVLY_START
 #endif
 
-#if XCHAL_CP_NUM > 0
```
Comment (xiaoxiang781216) [inline arch/xtensa/include/irq.h:126]: let's keep all place with reference XCHAL_CP_NUM except up_fpucmp

File: arch/xtensa/Kconfig (hunk 1 above)
Comment (Ouss4) [reply arch/xtensa/Kconfig:16]: Yes, ESP32 and ESP32-S3 both have an FPU coprocessor.  S3 has other coprocessors as well.

Comment (xiaoxiang781216) [reply arch/xtensa/Kconfig:16]: Ok, @zhuyanlinzyl please add riscv_fpucmp.c to Make.defs too

**Comment by zhuyanlinzyl** [comment]
hi, @Ouss4 : I update a new patch. CONFIG_ARCH_FPU only used for xtensa_fpucmp.c now. Please review again @acassis

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/xtensa/src/esp32s2/Make.defs [hunk 3]
```diff
@@ -52,6 +52,10 @@ ifeq ($(CONFIG_DEBUG_ALERT),y)
   CMN_CSRCS += xtensa_dumpstate.c
 endif
 
+ifeq ($(CONFIG_ARCH_FPU),y)
```
Comment (Ouss4) [inline arch/xtensa/src/esp32s2/Make.defs:55]: We can remove this one, ESP32-S2 does not have an FPU.

Comment (zhuyanlinzyl) [reply arch/xtensa/src/esp32s2/Make.defs]: Done.

**Review by Ouss4 — APPROVED** [review]
LGTM!

**Pull request merged**